src/util/createDomWrapper.ts

```diff
@@ -0,0 +1,72 @@
+import createWidget from '../createWidgetBase';
+import { Widget, WidgetProperties } from '../interfaces';
+import { VNodeProperties, VNode } from '@dojo/interfaces/vdom';
+import { ComposeFactory } from '@dojo/compose/compose';
+import WeakMap from '@dojo/shim/WeakMap';
+
+export interface DomWrapperProperties extends WidgetProperties {
+	domNode: Node;
+}
+
+export type DomWrapperFactory = ComposeFactory<Widget<DomWrapperProperties>, DomWrapperProperties>;
+
+interface DomWrapperPrivates {
+	vNode: VNode;
+}
+
+const domWrapperData = new WeakMap<Widget<DomWrapperProperties>, DomWrapperPrivates>();
+
+function handleDomInsertion(instance: Widget<DomWrapperProperties>, newNode: Node | null | undefined) {
+	let notNullNode = newNode;
+
+	if (!notNullNode) {
+		notNullNode = document.createElement('div'); // placeholder element
+	}
+
+	const data = domWrapperData.get(instance);
+
+	if (data) {
+		// replace the vNode domElement with our new element...
+		if (data.vNode.domNode && data.vNode.domNode.parentNode) {
+			data.vNode.domNode.parentNode.replaceChild(notNullNode, data.vNode.domNode);
+		}
+
+		// and update the reference to our vnode
+		data.vNode.domNode = notNullNode;
+	}
+}
+
+const createDomWrapper: DomWrapperFactory = createWidget.mixin({
+	mixin: {
+		afterCreate(this: Widget<DomWrapperProperties>) {
+			if (this.properties.domNode) {
+				// render has already been called at this point, so our vnode should be cached
+				const vNode = this.__render__();
+
+				if (vNode && typeof vNode !== 'string') {
+					// we need to hold on to our vNode for future updates
+					domWrapperData.set(this, {
+						vNode: vNode
+					});
+
+					handleDomInsertion(this, this.properties.domNode);
+				}
+			}
+		},
+		afterUpdate(this: Widget<DomWrapperProperties>) {
+			handleDomInsertion(this, this.properties.domNode);
+		},
+		nodeAttributes: [
```

Comment:
nodeAttributes will go away soon :), It will become an aspect after `render` to add the properties to the outer node